Two-body-scf-guess.py

```diff
@@ -0,0 +1,93 @@
+       ("H", 0.613411, 1.102620, 0.113724),
+       ("H", -0.093821, 2.209720, -0.643619),
+       ("O", 1.517569, -0.667424, -0.080674),
+       ("H", 1.989645, -1.098799, 0.612047),
+       ("H", 0.668397, -1.091798, -0.139744),
+       ("O", -1.350388, -0.964879, -0.092208),
+       ("H", -1.908991, -1.211298, 0.626207),
+       ("H", -1.263787, -0.018107, -0.055536)]
+
+bs = "aug-cc-pvdz"
+monomers = [mol[0:3], mol[3:6], mol[6:9]]
+dimers = [[mol[0:6], (mol[0:3]+mol[6:9])], [mol[0:6], mol[3:9]], [(mol[0:3]+mol[6:9]), mol[3:9]]]
+#dimers = [[(monomers[0], monomers[1]), (monomers[0], monomers[2]), (monomers[0], monomers[3])], 
+          #[(monomers[0], monomers[1]), (monomers[1], monomers[2]), (monomers[1], monomers[3])],
+          #[(monomers[0], monomers[2]), (monomers[1], monomers[2]), (monomers[2], monomers[3])],
+          #[(monomers[0], monomers[3]), (monomers[1], monomers[3]), (monomers[2], monomers[3])]]
+
+def make_monomer_dens(monomers, bs):
+    monomer_dens = []
+    for j in monomers:
+        j = density(j, bs)
+        monomer_dens.append(j)
+    return monomer_dens
+
+def make_list_of_ddls(dimers, bs):
+    list_of_ddls = []
+    for d_lists in dimers:
+        list_of_ddls.append([])
+        for dimer in d_lists:
+            dimer = density(dimer, bs)
+            list_of_ddls[-1].append(dimer)
+    return list_of_ddls
+
+    
+def make_rho():
+    rho = np.zeros((m*a, m*a))
+    return rho
+
+monomer_dens = make_monomer_dens(monomers, bs)
+m,m = monomer_dens[0].shape
+list_of_ddls = make_list_of_ddls(dimers, bs)
+a = len(monomer_dens)
+z_rho = make_rho()
+
+def add_interacts():
+    overlap = []
+    for i, dd_list in enumerate(list_of_ddls):
+        s = []
+        for j, dimer in enumerate(dd_list):
+            while j < i:
+                k = []
+                k.append(dimer)
+                bottom_right = sum(k)[m:m*2, m:m*2]
+                s.append(bottom_right)
+                break
+            while j >= i:
+                l = []
+                l.append(dimer)
+                top_left = sum(l)[0:m, 0:m]
+                s.append(top_left)
+                break
+        ss = sum(s)
+        overlap.append(ss)
+    return overlap 
+    
+def rho_diagonal():
+    overlap = add_interacts()
+    for i, dd_list in enumerate(list_of_ddls):
+        z_rho[i*m:(i+1)*m, i*m:(i+1)*m] = overlap[i]-(a-2)*monomer_dens[i]
+    return z_rho
+
+def rho_off_diag():
+    for i, dd_list in enumerate(list_of_ddls):
+        for j, dimer in enumerate(dd_list):
+            while j >= i:
+                z_rho[i*m:(i+1)*m, (j+1)*m:(j+2)*m] = dimer[0:m, m:m*2]
+                z_rho[(j+1)*m:(j+2)*m, i*m:(i+1)*m] = dimer[m:2*m, 0:m]
+                break
+    return z_rho
+    
+
+def two_body_guess(monomers, dimers, bs):
+    rho_d = rho_diagonal()
+    rho_o_d = rho_off_diag()
+    rho = np.add(rho_d, rho_o_d)
+    nitr = run_scf(mol, bs, rho)
+    return nitr 
+
+two_body_guess(monomers, dimers, bs)
```

Comment:
Unfortunately, I'm getting that the repo needs 14 cycles to converge. However inefficient, it seems to me that my process of doing this can work. It seems to me that either the dimer densities on rho are being incorrectly assigned, or that the diagonals aren't being correctly assigned. I don't know where these errors are happening, or how to fix them. I was hoping that you (Jonathan or Ryan) wouldn't mind taking a look to point out any line of code that is throwing off one or more of my functions. 

Right now I'm comparing rho's from the working script and this script to find where the errors are occurring.

Reply:
Literally I divided rho by 2 and the problem is solved.

Reply:
It is double what it should be for some reason.